### PR TITLE
feat(applications): Use the new `applications` CRD to store metadata about applications.

### PR DESCRIPTION
**Pull request by agentgonzo**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description
Store application metadata in the new `Application` CRD on a `jx import` (also `create quickstart` and `create spring`) and use this to get the organisation when doing a `jx delete app`

#### Special notes for the reviewer(s)
This stores additional metadata about the applications that can be used by other parts of the system. It also allows you to list applications that have been imported but _not_ deployed to staging/prod (eg, ones that are in the PR environment only) and also allows for quicker/better lookups of applications than trawling the `pipelineactivities` crd for _all_ things that jx has ever built.

My intention is that this CRD gets expanded with additional metadata that other devs need to store about applications.

Builds on #2642

### CODE REVIEW

File: pkg/kube/crds.go [hunk 1]
```diff
@@ -281,6 +285,46 @@ func RegisterAppCRD(apiClient apiextensionsclientset.Interface) error {
...
+			Name:        "Description",
+			Type:        "string",
+			Description: "A description of the source code repository - non-functional user-data",
+			JSONPath:    ".spec.description",
+		},
+		{
+			Name:        "Provider",
+			Type:        "string",
+			Description: "The source code provider (eg github) that the source repository is hosted in",
+			JSONPath:    ".spec.org",
```
Comment (jtnord) [inline pkg/kube/crds.go]: the `JSONPath` looks wrong to my eyes.

Comment (agentgonzo) [reply pkg/kube/crds.go]: fixed

**Comment by mikecirioli** [comment]
`serverless-jenkins` build is failing because this PR expects the `sourcerepositories` CRD to exist

**Pull request merged**